### PR TITLE
Added type declarations for the 'git-init' NPM package.

### PR DESCRIPTION
**Pull request by santi100a**
Please fill in this template.

- [] Use a meaningful title for the pull request. Include the name of the package modified.
- [] Test the change in your own code. (Compile and run.)
- [] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for git-init
```
Comment (peterblazejewicz) [inline index.d.ts:1]: what's this file for?
In doubt, please re-run:
```console
npx dts-gen --dt --name git-init --template module
```
to scaffold proper type declaration files structure for `git-init`:
https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package

File: types/git-init/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (peterblazejewicz) [inline types/git-init/LICENSE:1]: I believe the repo content is hosted and released under same license already:https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/LICENSE

File: types/git-init/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+declare function init(cb: Callback): void;
+/**
+ * Initializes a git repo asynchronously.
+ * @param path A path to the directory to initialize.
+ * @param cb An {@link exec} compatible callback.
+ */
+declare function init(path: string, cb: Callback): void;
+
+export default init;
```
Comment (peterblazejewicz) [inline types/git-init/index.d.ts:24]: the native module is 8 years old CommonJS file, it cannot be ESM with `default` export:
https://github.com/yoshuawuyts/git-init/blob/master/index.js#L3
please use generator in doubt, also review concept of CJS exports in TS:
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

File: types/git-init/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (peterblazejewicz) [inline types/git-init/package.json:1]: please remove this file, unless the intent was to add external dependencies.
In doubt, please re-run generate command as explained, to scaffold content

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
@santi100a thanks for the effort, please make amends as pointed in the comments, thanks!

**Review by gabritto — CHANGES_REQUESTED** [review]
Please fix what's been pointed out in the comments. Thanks.

**Comment by santi100a** [comment]
Superseded by #65055 and #65496.

**Pull request closed without merging**